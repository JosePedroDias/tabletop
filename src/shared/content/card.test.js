const random = require('../random').deterministic;

const card = require('./card')(random);

describe('card', () => {
  random.setSeed(42);
  const c1 = card.create();
  const c2 = card.create();
  const c3 = card.create({
    suit: 's',
    value: 'a',
    back: 'red',
    isFlipped: true
  });

  describe('constants', () => {
    it('SUITS', () => {
      expect(card.SUITS).toEqual(['h', 'd', 'c', 's']);
    });

    it('VALUES', () => {
      expect(card.VALUES).toEqual([
        '2',
        '3',
        '4',
        '5',
        '6',
        '7',
        '8',
        '9',
        '10',
        'j',
        'q',
        'k',
        'a'
      ]);
    });

    it('BACKS', () => {
      expect(card.BACKS).toEqual(['blue', 'green', 'red']);
    });
  });

  describe('create', () => {
    it('validate structure', () => {
      expect(c1).toMatchSnapshot();
      expect(c2).toMatchSnapshot();
      expect(c3).toMatchSnapshot();
    });

    it('validate suits', () => {
      expect(c1).toHaveProperty('data.suit', 'h');
      expect(c2).toHaveProperty('data.suit', 's');
      expect(c3).toHaveProperty('data.suit', 's');
    });

    it('validate values', () => {
      expect(c1).toHaveProperty('data.value', '4');
      expect(c2).toHaveProperty('data.value', '9');
      expect(c3).toHaveProperty('data.value', 'a');
    });
  });

  describe('actions', () => {
    it('flip()', () => {
      const cc1 = card.flip(c1);
      const cc2 = card.flip(c2);
      const cc3 = card.flip(c3);

      expect(cc1).toHaveProperty('data.isFlipped', true);
      expect(cc2).toHaveProperty('data.isFlipped', true);
      expect(cc3).toHaveProperty('data.isFlipped', false);

      expect(cc1).toMatchSnapshot();
      expect(cc2).toMatchSnapshot();
      expect(cc3).toMatchSnapshot();
    });
  });
});
