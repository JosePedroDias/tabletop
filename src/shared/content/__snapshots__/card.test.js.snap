// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card actions flip() 1`] = `
Object {
  "data": Object {
    "isFlipped": true,
    "isJoker": false,
    "suite": undefined,
    "value": "4",
  },
  "image": "assets/gfx/cards/back_red.png",
  "image2": "assets/gfx/cards/h4.png",
}
`;

exports[`card actions flip() 2`] = `
Object {
  "data": Object {
    "isFlipped": true,
    "isJoker": false,
    "suite": undefined,
    "value": "9",
  },
  "image": "assets/gfx/cards/back_blue.png",
  "image2": "assets/gfx/cards/s9.png",
}
`;

exports[`card actions flip() 3`] = `
Object {
  "data": Object {
    "isFlipped": false,
    "isJoker": false,
    "suite": undefined,
    "value": "a",
  },
  "image": "assets/gfx/cards/sa.png",
  "image2": "assets/gfx/cards/back_red.png",
}
`;

exports[`card create validate structure 1`] = `
Object {
  "data": Object {
    "isFlipped": false,
    "isJoker": false,
    "suit": "h",
    "value": "4",
  },
  "dimensions": Array [
    140,
    190,
  ],
  "image": "assets/gfx/cards/h4.png",
  "image2": "assets/gfx/cards/back_red.png",
  "kind": "card",
}
`;

exports[`card create validate structure 2`] = `
Object {
  "data": Object {
    "isFlipped": false,
    "isJoker": false,
    "suit": "s",
    "value": "9",
  },
  "dimensions": Array [
    140,
    190,
  ],
  "image": "assets/gfx/cards/s9.png",
  "image2": "assets/gfx/cards/back_blue.png",
  "kind": "card",
}
`;

exports[`card create validate structure 3`] = `
Object {
  "data": Object {
    "isFlipped": true,
    "isJoker": false,
    "suit": "s",
    "value": "a",
  },
  "dimensions": Array [
    140,
    190,
  ],
  "image": "assets/gfx/cards/back_red.png",
  "image2": "assets/gfx/cards/sa.png",
  "kind": "card",
}
`;
